tests/e2e-pw/specs/merchant/merchant-subscriptions-renew.spec.ts

```diff
@@ -0,0 +1,99 @@
+import {
+	describeif,
+	getMerchant,
+	getShopper,
+	prepareAcceptingDialogs,
+} from '../../utils/helpers';
+import {
+	emptyCart,
+	ensureSavedCardNotSelected,
+	fillCardDetails,
+	focusPlaceOrderButton,
+	placeOrder,
+	setupProductCheckout,
+} from '../../utils/shopper';
+import { goToShop } from '../../utils/shopper-navigation';
+import { goToSubscriptionPage } from '../../utils/merchant-navigation';
+import { shouldRunSubscriptionsTests } from '../../utils/constants';
+
+const productName = 'Subscription signup fee product';
+const customerBillingConfig =
+	config.addresses[ 'subscriptions-customer' ].billing;
+let subscriptionId = null;
+
+describeif( shouldRunSubscriptionsTests )( 'Admin order analytics', () => {
+	test.beforeAll( async ( { browser }, { project } ) => {
+		const restApi = new RestAPI( project.use.baseURL );
+		restApi.deleteCustomerByEmailAddress( customerBillingConfig.email );
+
+		const { shopperPage } = await getShopper( browser );
+		await emptyCart( shopperPage );
+		await goToShop( shopperPage, 2 );
+		await setupProductCheckout(
+			shopperPage,
+			[ [ productName, 1 ] ],
+			customerBillingConfig
+		);
+		await ensureSavedCardNotSelected( shopperPage );
+		await fillCardDetails( shopperPage, config.cards.basic );
+		await focusPlaceOrderButton( shopperPage );
+		await placeOrder( shopperPage );
+		await shopperPage.waitForURL( /\/order-received\//, {
+			waitUntil: 'load',
+		} );
+		await expect(
+			shopperPage.getByRole( 'heading', { name: 'Order received' } )
+		).toBeVisible();
+
+		// Get the subscription ID
+		const subscriptionIdField = await shopperPage.locator(
+			'.woocommerce-orders-table__cell-subscription-id > a'
+		);
+		subscriptionId = await subscriptionIdField.evaluate( ( el ) =>
+			el.textContent.trim().replace( '#', '' )
+		);
+		// Give some time for the subscription to be created
+		await shopperPage.waitForLoadState( 'networkidle' );
+	} );
+
+	test( 'should be able to renew a subscription in my account', async ( {
+		browser,
+	} ) => {
+		const { merchantPage } = await getMerchant( browser );
+		await goToSubscriptionPage( merchantPage, subscriptionId );
+		await expect(
+			merchantPage.getByRole( 'heading', {
+				name: 'Edit Subscription',
+				level: 1,
+			} )
+		).toBeVisible();
+		const orderActions = await merchantPage.locator(
+			'select[name="wc_order_action"]'
+		);
+		await orderActions.selectOption( { label: 'Process renewal' } );
+
+		// Prepare to accept the dialog before clicking the submit button.
+		// Since the page will change, we don't need to remove the listener.
+		prepareAcceptingDialogs( merchantPage );
+		await merchantPage.click( 'button:has-text("Apply")' );
+		await merchantPage.waitForLoadState( 'networkidle' );
+
+		// Check if a new order is present in related orders
+		await expect(
+			merchantPage.locator(
+				'div.woocommerce_subscriptions_related_orders > table > tbody > tr > td',
+				{ hasText: 'Renewal Order' }
+			)
```

Comment:
Using a long selector like that can be quite fragile, we can just check for a cell with the desired text.

```suggestion
			merchantPage.getByRole( 'cell', {
				name: 'Renewal Order',
			} )
```

Reply:
Replaced in [405bbbf](https://github.com/Automattic/woocommerce-payments/pull/10153/commits/405bbbfceeaef2247fa9fee947e4eededf2a9861).